File: schemas/podcast-platform/mongodb/transcripts.js
// transcripts: Episode transcripts in multiple formats with optional content storage.
// See README.md for full design rationale.
const mongoose = require("mongoose");

const transcriptsSchema = new mongoose.Schema(
  {
    episode_id: { type: mongoose.Schema.Types.ObjectId, ref: "Episode", required: true },
    format: { type: String, enum: ["srt", "vtt", "json", "text"], required: true },
    content_url: { type: String, default: null },
    content: { type: String, default: null },
    language: { type: String, required: true, default: "en" },
    is_auto_generated: { type: Boolean, required: true, default: false },
  },
  { timestamps: { createdAt: "created_at", updatedAt: "updated_at" } }
);

transcriptsSchema.index({ episode_id: 1, format: 1, language: 1 });

module.exports = mongoose.model("Transcript", transcriptsSchema);

File: schemas/podcast-platform/spacetimedb/transcripts.rs
// transcripts: Episode transcript content in various formats and languages.
// See README.md for full design rationale.

use spacetimedb::SpacetimeType;
use spacetimedb::Timestamp;

#[derive(SpacetimeType, Clone)]
pub enum TranscriptFormat {
    Srt, // type: String
    Vtt,
    Json,
    Text,
}

#[spacetimedb::table(name = transcripts, public)]
pub struct Transcript {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub episode_id: String, // UUID — FK → episodes.id (cascade delete)
    pub format: TranscriptFormat,
    pub content_url: Option<String>,
    pub content: Option<String>,
    pub language: String,
    pub is_auto_generated: bool,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
// Composite index: (episode_id, format, language) — not supported, enforce in reducer logic
